### ISSUE TITLE
replicate is not installed automatically and throw exception

### ISSUE THREAD

**Issue by yujonglee** [BerriAI/litellm#278]
What's the problem? (if there are multiple - list as bullet points)

<img width="975" alt="Screenshot 2023-09-06 at 7 56 04 PM" src="https://github.com/BerriAI/litellm/assets/61503739/5334ccbe-e1bb-4110-8775-72c7da35bc27">

To reproduce: use replicate with `litellm.gpt_cache.completion`.

https://github.com/BerriAI/litellm/blob/64f3d3c56ef02ac5544983efc78293de31c1c201/litellm/main.py#L344

Seems like this does not work well with Replicate? Not sure.
The error disappears when I install replicate.

**Comment by ishaan-jaff** [comment]
@yujonglee 
we've deprecated install_and_import in the latest version of litellm due to the above mentioned issues 

now we try importing replicate in try/except. If it fails we ask users to pip install replicate

**Comment by ishaan-jaff** [comment]
see https://github.com/BerriAI/litellm/blob/main/litellm/main.py#L344 

```
elif "replicate" in model or custom_llm_provider == "replicate":
            # import replicate/if it fails then pip install replicate
            try:
                import replicate
            except:
                Exception("Replicate import failed please run `pip install replicate`")
```

**Comment by ishaan-jaff** [comment]
@yujonglee what do you think of this ? We don't want to add `replicate` as a dependency to litellm

**Comment by ishaan-jaff** [comment]
perhaps the resolution here can just be to stop using the replicate python package and just call their API directly

**Comment by ishaan-jaff** [comment]
here's how it works using http requests, let's start using this 


```
# Define the URL for making predictions
url = "https://api.replicate.com/v1/predictions"

# Define the headers with the authorization token
headers = {
    "Authorization": f"Token {os.environ['REPLICATE_API_TOKEN']}"
}

# Define the JSON data for the initial prediction request
data = {
    "version": "1c914d844307b0588599b8393480a3ba917b660c7e9dfae681542b5325f228db",
    "input": {
        "prompt": "who are you?"
    }
}

# Send the initial POST request to start the prediction
response = requests.post(url, json=data, headers=headers)

# Check if the request was successful
if response.status_code == 201:
    # Parse the JSON response
    response_data = response.json()
    print("Initial Prediction Response:")
    print(json.dumps(response_data, indent=2))

    # Check the status of the prediction
    while response_data["status"] != "succeeded":
        if response_data["status"] == "failed":
            print("Prediction failed.")
            break

        # Fetch the prediction status and output
        prediction_url = f"{url}/{response_data['id']}"
        response = requests.get(prediction_url, headers=headers)
        
        if response.status_code == 200:
            prediction_data = response.json()
            print("Prediction Status and Output:")
            print(json.dumps(prediction_data, indent=2))
            response_data = prediction_data  # Update response_data for the next iteration
        else:
            print("Failed to fetch prediction status and output.")

        time.sleep(0.0001)
else:
    print("Initial prediction request failed with status code:", response.status_code)
```

**Comment by ishaan-jaff** [comment]
started working on this here: https://github.com/BerriAI/litellm/commit/c45b1326753c8f2ce105eaef341a2bad8fdce13b

**Comment by ishaan-jaff** [comment]
cc @oliverfeng @crazycth for awareness
TLDR: We're updating replicate to directly make replicate https requests instead of using the replicate py package

**Comment by ishaan-jaff** [comment]
marking this as completed, we've moved replicate to https requests instead of using the package
